w09h03/test/pgdp/filter/DGTest.java

```diff
@@ -0,0 +1,5539 @@
+                Assertions.fail();
+            }
+        }
+
+        int[] colorsE = new int[]{Color.BLACK.getRGB(), Color.WHITE.getRGB(), Color.WHITE.getRGB(),
+                Color.BLACK.getRGB(), Color.BLACK.getRGB(), Color.WHITE.getRGB(), Color.BLACK.getRGB(), Color.WHITE.getRGB()};
+
+        for(int i = 0; i < amount; i++){
+            Assertions.assertEquals(width, output[i].getWidth());
+            Assertions.assertEquals(height, output[i].getHeight());
+            Assertions.assertEquals(numbers[i], output[i].getFrameNumber());
+            if(i < amount){
+                // encoding path
+                for(int j = 0; j < 8; j++){
+                    Assertions.assertEquals(colorsE[j], output[i].getPixels().getRGB(j,output[i].getHeight()-1));
+                    if(i < amount){
+                        Assertions.assertEquals(colorsE[j], output[i].getPixels().getRGB(8+j,output[i].getHeight()-1));
+                    }
+                }
+                // rest
+                for(int j = 8; j < output[i].getWidth(); j++){
+                    if(i >= amount){
+                        Assertions.assertEquals(Color.YELLOW.getRGB(), output[i].getPixels().getRGB(j,output[i].getHeight()-1));
+                    }
+                }
+                for(int j = 0; j < output[i].getWidth(); j++){
+                    for(int k = 0; k < output[i].getHeight() - 1; k++){
+                        Assertions.assertEquals(Color.YELLOW.getRGB(), output[i].getPixels().getRGB(j,k));
+                    }
+                }
+            }
+            else{
+                for(int j = 0; j < output[i].getWidth(); j++){
+                    for(int k = 0; k < output[i].getHeight(); k++){
+                        Assertions.assertEquals(Color.YELLOW.getRGB(), output[i].getPixels().getRGB(j,k),"failed at frame "+i+" at index "+j+","+k);
+                    }
+                }
+            }
+
+        }
+
+    }
+
+    @Test
+    @DisplayName("Test DG 39 Encode Stream 5 frames 16x16 mit null")
+    public void test39() throws IOException {
+        int size = 16;
+        int amount = 5;
+        BufferedImage[] images = new BufferedImage[amount];
+        for(int k = 0; k < amount; k++){
+            images[k] = new BufferedImage(size,size,BufferedImage.TYPE_INT_ARGB);
+            for(int i = 0; i < size; i++){
+                for(int j = 0; j < size; j++){
+                    images[k].setRGB(i,j,Color.YELLOW.getRGB());
+                }
+            }
+        }
+
+
+        int height = images[0].getHeight();
+        int width = images[0].getWidth();
+        Frame[] frames = new Frame[amount];
+        int[] numbers = new int[amount];
+        for(int i = 0; i < frames.length; i++){
+            frames[i] = new Frame(images[i], i);
+            numbers[i] = i;
+        }
+        Stream<Frame> frameStream = Arrays.stream(frames);
+
+        Function<Frame, Frame> function = Operations.encode(null);
```

Comment:
Wrap this in a try, catch block. My encode method requires a nonNull object and throws an IllegalArgumentException if the argument is null (Most of the java base library behaves in a similar manner).

Reply:
I think you cannot necessarily argue for your function, yes the argument should never be null, but we know from the past that there have been tests. Further, we know our code shall never crash, except required, thus I'd rather go with Dominiks PR, than throwing an exception which was definitely not asked for!